pkg/apis/operator/v1alpha1/common.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+// CommonSpec unifies common fields and functions on the Spec.
+type CommonSpec struct {
+	// A means to override the corresponding entries in the upstream configmaps
+	// +optional
+	Config map[string]map[string]string `json:"config,omitempty"`
+
+	// A means to override the corresponding deployment images in the upstream.
+	// If no registry is provided, the knative release images will be used.
+	// +optional
+	Registry Registry `json:"registry,omitempty"`
+
+	// Override containers' resource requirements
+	// +optional
+	Resources []ResourceRequirementsOverride `json:"resources,omitempty"`
+}
+
+// Config represents a map of upstream config map data.
+type Config map[string]map[string]string
```

Comment:
Is this line still useful?

Reply:
good point, I should hook it up with the above.

Reply:
Fixed, took it out.